sharding/networksharding/peerShardMapper.go

```diff
@@ -191,9 +199,27 @@ func (psm *PeerShardMapper) getPeerInfoSearchingPidInFallbackCache(pid core.Peer
 		}
 	}
 
+	subTypeObj, ok := psm.peerIdSubType.Get([]byte(pid))
+	if !ok {
+		return &core.P2PPeerInfo{
+			PeerType: core.ObserverPeer,
+			ShardID:  shard,
+		}
+	}
+	subType, ok := subTypeObj.(uint32)
```

Comment:
why not casting directly to core.P2PPeerSubType ?

Reply:
Done